### DISCUSSION TITLE
Window disappears when powering off monitor

### DISCUSSION THREAD

**Discussion by trames57** [WayfireWM/wayfire#2341] [Q&A] [answered]
I have an older PDF reader I use called Foxit Reader. When I launch the program, everything works OK and the icon shows up in the Waybar tray.  When I power off the monitor and later power it back on, the window of the app and the tray icon disappear. The process however is still running when executing the ps command.  I do have the preserve-output plugin running.

Am I missing a config option or plugin?

Type: desktop, one monitor
Distro: Arch Linux
Wayfire: 0.8.1
Waybar: 0.9.24

lswt output:
`0: ---- "Start - Foxit Reader" "Foxit Reader"`

**Comment by ammen99** [comment]
Is this PDF viewer the only app which has a problem like this, or does it happen with other applications as well?

**Reply by trames57** [reply]
Other apps that I use on a regular basis retain their windows such as
Google Chrome
GIMP 2.10
GNUCash
xfce4-terminal
Jetbrains - Webstorm IDE

**Reply by ammen99** [reply]
Ok, could you check whether the app maybe doesn't unmap its window? Also, for debugging this it would be good to gather some debug logs, try `wayfire -d -d txn -d txni -d views -d xwayland &> log.txt` reproduce the issue (try to do as few operations as possible) and then upload the log file.

**Reply by trames57** [reply]
I am not sure how to test for "unmap". If you mean does lswt show the window after I power the monitor back on, then no, it is not in the list. I uploaded the log file in a new comment.

**Comment by trames57** [comment]
Wayfire debug log. You can scan for Foxit.  I powered off the monitor at 11:11:48.  I turned it back on at 11:17:13.
I do see xcb errors, but do not know what those mean.

[wayfire_log.txt](https://github.com/WayfireWM/wayfire/files/15052499/wayfire_log.txt)

**Reply by ammen99** [reply, marked as answer]
trames57, the log indicates that the pdf viewer unmaps itself when the output is disconnected. This means that it hides its surface and should not be shown to the user at all (which Wayfire correctly does). This could be a bug in the application, or maybe it hides itself in the systray (I have seen other apps which do that) or maybe you need to run the application again and it will unmap itself. There is unfortunately not much that we can do about it on the Wayfire side - the app decides to hide itself on its own (unmap is not something that Wayfire can do, the client has to request it).

**Reply by trames57** [reply]
It worked with the Xserver just fine of course using the i3 window manager. I recently converted from that to Wayfire in March. I do understand that XWayland is not quite Xserver and some things just won't work the same.  Foxit Reader will not launch again if it finds the same app already running, so I have to pkill it.

I will seek out a newer PDF viewer that has a comparable annotation implementation. Thank you so much for your time and attention!
